apps/maestro/src/server/routers/sui/index.ts

```diff
@@ -183,14 +184,33 @@ export const suiRouter = router({
         // Split coins for gas
         const Gas = tx.splitCoins(tx.gas, [BigInt(input.gas)]);
 
-        const coins = await suiClient.getCoins({
-          owner: input.sender,
-          coinType: input.coinType,
-        });
-        const coinObjectId = coins.data[0].coinObjectId;
+        // Get all coins of the specified type
+        let primaryCoin: string = "";
+        let coins: PaginatedCoins;
+        let otherCoins: string[] = [];
+        do {
+          coins = await suiClient.getCoins({
+            owner: input.sender,
+            coinType: input.coinType,
+          });
+
+          if (coins.data.length === 0) {
+            throw new Error("No coins found");
+          }
+
+          // If there are multiple coins, merge them first
+          primaryCoin = !primaryCoin ? coins.data[0].coinObjectId : primaryCoin;
```

Comment:
This should only be set once instead of reassigning the value for every paginated result.